File: HelloWorld/task3/GraphBfsDfsJava/src/com/graph/Graph.java
package com.graph;

import java.io.*;
import java.util.*;

public class Graph {
    private int _V;
    private LinkedList<Integer> _adj[];

    public Graph(int v) {
        _V = v;
        _adj = new LinkedList[v];
        for (int i = 0; i < v; ++i)
            _adj[i] = new LinkedList();
    }

    public void addEdge(int v, int w) {
        _adj[v].add(w);
    }

    public void BFS(int s) {
        boolean visited[] = new boolean[_V];

        LinkedList<Integer> queue = new LinkedList<Integer>();

        visited[s] = true;
        queue.add(s);

        while (queue.size() != 0) {
            s = queue.poll();
            System.out.print(s + " ");

            Iterator<Integer> i = _adj[s].listIterator();
            while (i.hasNext()) {
                int n = i.next();
                if (!visited[n]) {
                    visited[n] = true;
                    queue.add(n);
                }
            }
        }
    }

    private void DFSUtil(int v, boolean visited[]) {
        visited[v] = true;
        System.out.print(v + " ");

        Iterator<Integer> i = _adj[v].listIterator();
        while (i.hasNext()) {
            int n = i.next();
            if (!visited[n])
                DFSUtil(n, visited);
        }
    }

    public void DFS(int v) {
        boolean visited[] = new boolean[_V];
        DFSUtil(v, visited);
    }
}

File: HelloWorld/task3/GraphBfsDfs/GraphBfsDfs/Graph.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace GraphBfsDfs
{
    public class Graph
    {
        private int _V; 
        private LinkedList<int>[] _adj;
 
        public Graph(int v)
        {
            _adj = new LinkedList<int>[v];
            for(int i = 0; i < _adj.Length; i++)
            {
                _adj[i] = new LinkedList<int>();
            }
            _V = v;
        }
 
        public void AddEdge(int v, int w)
        {
            _adj[v].AddLast(w);
        }
        
        public void BFS(int s)
        {
            bool[] visited = new bool[_V];
            for (int i = 0; i < _V; i++)
                visited[i] = false;

            LinkedList<int> queue = new LinkedList<int>();

            visited[s] = true;
            queue.AddLast(s);

            while (queue.Any())
            {
                s = queue.First();
                Console.Write(s + " ");
                queue.RemoveFirst();

                LinkedList<int> list = _adj[s];

                foreach (var val in list)
                {
                    if (!visited[val])
                    {
                        visited[val] = true;
                        queue.AddLast(val);
                    }
                }
            }
        }
        private void DFSUtil(int v, bool[] visited)
        {
            visited[v] = true;
            Console.Write(v + " ");
 
            LinkedList<int> vList = _adj[v];
            foreach(var n in vList)
            {
                if (!visited[n])
                    DFSUtil(n, visited);
            }
        }
 
        public void DFS(int v)
        {
            bool[] visited = new bool[_V];
            DFSUtil(v, visited);
        }
    }
}